### ISSUE TITLE
cl.find_label_issues usage

### ISSUE THREAD

**Issue by Solenyalyl** [cleanlab/cleanlab#1158]
hello,
I have used 
cl=cleanlab.classification.CleanLearning(model,seed=42)
then cl.find_label_issues()  in train,validation and test data,when the label is judged 'True' in 'is_label_issue' column in output file, it was deleleted in the next train ,and it always get a good result, I'm very confused about it, It's alright to get such a good result in train validation and test result ,since we haven't gotten more test in real world data.

**Comment by Solenyalyl** [comment]
got it, thank you


| |
刘雨露
生信工程师
|
|
万基生物
18559760807
***@***.***
中国科学院大学华大教育中心硕士
|
---- Replied Message ----
| From | Elías ***@***.***> |
| Date | 6/24/2024 23:50 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [cleanlab/cleanlab] cl.find_label_issues usage (Issue #1158) |

Hi @Solenyalyl!

Thanks for reaching out!

I have used
cl=cleanlab.classification.CleanLearning(model,seed=42)
then cl.find_label_issues() in train,validation and test data,when the label is judged 'True' in 'is_label_issue' column in output file, it was deleleted in the next train ,and it always get a good result, I'm very confused about it

It seems you're applying CleanLearning to your entire dataset (train, validation, and test) and then removing all identified issues before training. It can be problematic for several reasons:

Automatically removing issues from your test set can lead to overly optimistic performance estimates.
This method might not generalize well to real-world data that hasn't been 'cleaned' in the same way.
By applying the same cleaning process to both training and test data, you may be introducing data leakage.

It's alright to get such a good result in train validation and test result ,since we haven't gotten more test in real world data.

It's important to be careful with your training & test data to ensure reliable results. The basic approach with CleanLearning goes as follows:

Use CleanLearning to find label issues in the entire dataset:

fromcleanlab.classificationimportCleanLearningcl=CleanLearning(your_model, seed=42)
label_issues=cl.find_label_issues(X=all_data, labels=all_labels)

For test data:

Review the identified issues manually. Only make changes if you are very sure they will make the evaluation more reliable.
Be cautious about removing examples. Only remove data if you are certain it does not represent real-world scenarios.

For training data:

Use CleanLearning to train a robust model:
cl.fit(X=train_data, labels=train_labels)
This will automatically handle the identified issues in the training data.

Evaluate the model on your manually reviewed test set:

predictions=cl.predict(test_data)

This approach leverages CleanLearning to identify issues across your entire dataset and automatically handle them in the training process, while still allowing for careful manual review of test data.

If you are not automatically removing issues from the test set, that's good! Just ensure you follow the careful manual review process to maintain reliable evaluation.

Is this process clear?

**Comment by Solenyalyl** [comment]
![963](https://github.com/cleanlab/cleanlab/assets/33305699/191daa0a-6933-4e17-9927-e3dae6a5ba14)
here x=all_data,the all_data includes test data,right?

**Comment by elisno** [comment]
> here `X=all_data`, the `all_data` includes test data, right?

Yes, `all_data` can include test data for detecting label issues. This allows `CleanLearning` to identify problems in the test data using as much information as possible. Just remember to manually review and handle any issues in the test data to ensure a reliable evaluation.

**Comment by Solenyalyl** [comment]
thanks



---- Replied Message ----
| From | Elías ***@***.***> |
| Date | 06/26/2024 18:57 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [cleanlab/cleanlab] cl.find_label_issues usage (Issue #1158) |

here X=all_data, the all_data includes test data, right?

Yes, all_data can include test data for detecting label issues. This allows CleanLearning to identify problems in the test data using as much information as possible. Just remember to manually review and handle any issues in the test data to ensure a reliable evaluation.
